### PR TITLE
Allow a steam return url to be passed through

### PR DESCRIPTION
**Pull request by atoff**
Sometimes (I have found in my application, anyway) it is useful to push the user from one url, and have them return to another one that is not the same as that specified in the config.

Simply allows a URL to be passed through when constructing the SteamAuth class.

### CODE REVIEW

**Comment by nerdalertdk** [comment]
Please yes

File: README.md [hunk 1]
```diff
@@ -160,3 +160,12 @@ class AuthController extends Controller
 }
 
 ```
+
+Should you wish to use a login redirection URL that is differant from the one you specified in the config
+
+```
```
Comment (Gummibeer) [inline README.md]: `php` as code-type is missing

File: README.md [hunk 2]
```diff
@@ -160,3 +160,12 @@ class AuthController extends Controller
...
+    public function __construct(Request $request)
```
Comment (Gummibeer) [inline README.md]: the indent is not used here - it's the first code-level in this snippet.

File: src/SteamAuth.php [hunk 3]
```diff
@@ -49,10 +49,13 @@ class SteamAuth implements SteamAuthInterface
      *
      * @param Request $request
      */
-    public function __construct(Request $request)
+    public function __construct(Request $request, $redirect_url = null)
```
Comment (Gummibeer) [inline src/SteamAuth.php]: This class is bind as singleton in https://github.com/invisnik/laravel-steam-auth/blob/master/src/SteamServiceProvider.php#L32
If you want to customize the redirect-url you should adjust the `redirect` method.

```php
public function redirect($redirectUrl = null)
{
    return redirect(is_null(redirectUrl) ? $this->getAuthUrl() : $this->buildUrl(url($redirectUrl)));
}
```

**Comment by Gummibeer** [comment]
@atoff thanks for adjusting, in laravel it's common to return `$this` in setter methods.
With this it would be possible to do `$steam->setRedirectUrl(...)->redirect()`.

**Pull request merged**